src/controller_examples/rabbitmq_controller/proof/liveness/resource_match.rs

```diff
@@ -529,21 +699,27 @@ proof fn lemma_resource_state_matches_at_after_create_resource_step(
         lift_state(RMQCluster::busy_disabled()),
         lift_state(RMQCluster::every_in_flight_msg_has_unique_id()),
         lift_state(RMQCluster::each_object_in_etcd_is_well_formed()),
+        lift_state(helper_invariants::triggering_cr_satisfies_state_validation()),
         lift_state(helper_invariants::cm_rv_is_the_same_as_etcd_server_cm_if_cm_updated(rabbitmq)),
         lift_state(helper_invariants::every_resource_create_request_implies_at_after_create_resource_step(SubResource::ServerConfigMap, rabbitmq)),
         lift_state(helper_invariants::every_resource_create_request_implies_at_after_create_resource_step(sub_resource, rabbitmq))
     );
 
-    let post = current_state_matches(sub_resource, rabbitmq);
+    let post = |s: RMQCluster| {
+        &&& sub_resource_state_matches(sub_resource, rabbitmq)(s)
+        &&& at_after_create_resource_step_and_exists_ok_resp_in_flight(sub_resource, rabbitmq)(s)
+    };
 
     assert forall |s, s_prime: RMQCluster| pre(s) && #[trigger] stronger_next(s, s_prime) && RMQCluster::kubernetes_api_next().forward(input)(s, s_prime) implies post(s_prime) by {
-        assert(make(sub_resource, rabbitmq, s.ongoing_reconciles()[rabbitmq.object_ref()].local_state).is_Ok());
+        let pending_msg = s.ongoing_reconciles()[rabbitmq.object_ref()].pending_req_msg.get_Some_0();
+        let resp = RMQCluster::handle_create_request(pending_msg, s.kubernetes_api_state).1;
+        assert(s_prime.in_flight().contains(resp));
         match sub_resource {
```

Comment:
This pattern matching seems to appear multiple times. You might consider fold it into a macro if it is worthwhile.